ui/admin/tests/acceptance/credential-store/credentials/update-test.js

```diff
@@ -128,57 +224,128 @@ module(
       await click('form [type="button"]', 'Activate edit mode');
       await fillIn('[name="name"]', 'random string');
-      assert.strictEqual(currentURL(), urls.credential);
+      assert.strictEqual(currentURL(), urls.usernamePasswordCredential);
       try {
         await visit(urls.credentials);
       } catch (e) {
         assert.ok(find('.rose-dialog'));
         await click('.rose-dialog-footer button:first-child', 'Click Discard');
         assert.strictEqual(currentURL(), urls.credentials);
         assert.notEqual(
-          this.server.schema.credentials.all().models[0].name,
+          this.server.schema.credentials.where({ type: 'username_password' })
+            .models[0].name,
           'random string'
         );
       }
     });
 
-    test('can cancel discard unsaved credential changes via dialog', async function (assert) {
+    test('can discard unsaved username & key pair credential changes via dialog', async function (assert) {
       assert.expect(5);
       const confirmService = this.owner.lookup('service:confirm');
       confirmService.enabled = true;
-      assert.notEqual(instances.credential.name, 'random string');
-      await visit(urls.credential);
+      assert.notEqual(
+        instances.usernameKeyPairCredential.name,
+        'random string'
+      );
+      await visit(urls.usernameKeyPairCredential);
       await click('form [type="button"]', 'Activate edit mode');
       await fillIn('[name="name"]', 'random string');
-      assert.strictEqual(currentURL(), urls.credential);
+      assert.strictEqual(currentURL(), urls.usernameKeyPairCredential);
+      try {
+        await visit(urls.credentials);
+      } catch (e) {
+        assert.ok(find('.rose-dialog'));
+        await click('.rose-dialog-footer button:first-child', 'Click Discard');
+        assert.strictEqual(currentURL(), urls.credentials);
+        assert.notEqual(
+          this.server.schema.credentials.where({ type: 'ssh_private_key' })
+            .models[0].name,
+          'random string'
+        );
+      }
+    });
+
+    test('can cancel discard unsaved username & password credential changes via dialog', async function (assert) {
+      assert.expect(5);
+      const confirmService = this.owner.lookup('service:confirm');
+      confirmService.enabled = true;
+      assert.notEqual(
+        instances.usernamePasswordCredential.name,
+        'random string'
+      );
+      await visit(urls.usernamePasswordCredential);
+      await click('form [type="button"]', 'Activate edit mode');
+      await fillIn('[name="name"]', 'random string');
+      assert.strictEqual(currentURL(), urls.usernamePasswordCredential);
+      try {
+        await visit(urls.credentials);
+      } catch (e) {
+        assert.ok(find('.rose-dialog'));
+        await click('.rose-dialog-footer button:last-child', 'Click Cancel');
+        assert.strictEqual(currentURL(), urls.usernamePasswordCredential);
+        assert.notEqual(
```

Comment:
Perhaps I'm not understanding this test, but what exactly are we testing for here? Shouldn't we check the field to be `random string` still to show it wasn't discarded? And check that the model in the schema is still the same value as it hasn't been saved yet?

Reply:
This test is mainly to test that the cancel button works when trying to click away from a form that was being edited. Then it tests that the changes aren't in the schema of the credential because the form hasn't been saved. However, I do agree with your suggestion, I will add an assert statement for that.